client/app/scripts/superdesk-desks/desks.js

```diff
@@ -73,29 +73,43 @@
                 allowed: '=',
                 showEmpty: '=?',
                 selected: '=?',
-                action: '&'
+                action: '&',
+                filter: '='
             },
             link: function(scope, elem) {
 
-                var query = search.query({});
-                query.filter({term: {'task.stage': scope.stage}});
-                query.size(10);
-                var criteria = {source: query.getCriteria()};
-
-                scope.loading = true;
-
-                api('archive').query(criteria).then(function(items) {
-                    scope.loading = false;
-                    scope.items = items._items;
-                    scope.total = items._meta.total;
-                }, function() {
-                    scope.loading = false;
-                });
-
                 scope.open = function(item) {
                     desks.setWorkspace(item.task.desk, item.task.stage);
                     superdesk.intent('read_only', 'content_article', item);
                 };
+
+                function queryItems(queryString) {
+                    var query = search.query({});
+                    query.filter({term: {'task.stage': scope.stage}});
+                    query.size(10);
+
+                    if (queryString) {
+                        query.filter({query: {query_string: {
+                            query: queryString,
+                            lenient: false
+                        }}});
+                    }
+
+                    var criteria = {source: query.getCriteria()};
+
+                    scope.loading = true;
+                    scope.items = scope.total = null;
+                    api('archive').query(criteria).then(function(items) {
+                        scope.loading = false;
+                        scope.items = items._items;
+                        scope.total = items._meta.total;
+                    }, function() {
+                        scope.loading = false;
```

Comment:
This `scope.loading` repeats in both the success and the error handler, better to move it into `finally()` to avoid code repetition.